api/client/src/main/java/org/projectnessie/client/rest/ResponseCheckFilter.java

```diff
@@ -84,52 +86,74 @@ public static void checkResponse(ResponseContext con) throws Exception {
         exception = new NessieServiceException(error);
     }
 
-    if (error.getClientProcessingException() != null) {
-      exception.addSuppressed(error.getClientProcessingException()); // for trouble-shooting
-    }
-
     throw exception;
   }
 
   private static NessieError decodeErrorObject(
-      Status status, InputStream inputStream, ObjectReader reader) {
-    NessieError error;
+      Status status, ResponseContext requestCtx, InputStream inputStream, ObjectReader reader) {
+    ImmutableNessieError.Builder error =
+        ImmutableNessieError.builder()
+            .status(status.getCode())
+            .reason(status.getReason())
+            .message(
+                requestCtx.getRequestedMethod() + " " + requestCtx.getRequestedUri() + " failed");
```

Comment:
Why pre-build this when most error responses will use the server-provided error object on line 137?

Reply:
It's an object that will need to be GC'd... even if efficiently :)